omc3/correction/response_twiss.py

```diff
@@ -0,0 +1,875 @@
+    \frac{cos(\tau_{y,mj})}{sin(\pi Q_y)}
+
+
+* Norm. Dispersion Response: similar as above but with :math:`\frac{1}{\sqrt{\beta}}` linearized
+
+.. math::
+
+    \delta \frac{D_{x,j}}{\sqrt{\beta_{x,j}}} =&+ \sum_m (\delta K_{0,m} + \delta K_{1S,m} D_{y,m}
+    - \delta K_{1,m} D_{x,m} ) \frac{\sqrt{\beta_{x,m}}}{2}
+    \frac{cos(\tau_{x,mj})}{sin(\pi Q_x)}
+    &&+ \frac{D_{x,j}}{\sqrt{\beta_{x,j}}} \delta K_{1,m}
+    \frac{\beta_{x,m}}{4}\frac{cos(2\tau_{x,mj})}{2sin(\pi Q_x)}
+    \\
+    \delta \frac{D_{y,j}}{\sqrt{\beta_{y,j}}} =&- \sum_m (\delta K_{0S,m} - \delta K_{1S,m} D_{x,m}
+    - \delta K_{1,m} D_{y,m}) \frac{\sqrt{\beta_{y,m}}}{2}
+    \frac{cos(\tau_{y,mj})}{sin(\pi Q_y)}
+    &&- \frac{D_{y,j}}{\sqrt{\beta_{y,j}}} \delta K_{1,m}
+    \frac{\beta_{y,m}}{4}\frac{cos(2\tau_{y,mj})}{2sin(\pi Q_y)}
+
+
+* Phase Advance Response:    Eq. 28 in [#FranchiAnalyticformulasrapid2017]_
+
+.. math::
+
+    \delta \Phi_{z,wj} = \pm \sum_m \delta K_{1,m} \frac{\beta_{z,m}}{4}
+    \left\{ 2\left[ \Pi_{mj} - \Pi_{mw} + \Pi_{jw} \right] +
+    \frac{sin(2\tau_{z,mj}) - sin(2\tau_{z,mw})}{sin(2\pi Q_z)} \right\}
+
+
+* Tune Response:             Eq. 7 in [#TomasReviewlinearoptics2017]_
+
+.. math::
+
+    \delta Q_z = \pm \sum_m \delta K_{1,m} \frac{\beta_{z,m}}{4\pi}
+
+
+* Coupling Response:            Eq. 10 in [#FranchiAnalyticformulasrapid2017]_
+
+.. math::
+
+    \delta f_{\substack{\scriptscriptstyle 1001 \\ \scriptscriptstyle 1010},j} =
+    \sum_m \delta J_{1,m} \, \frac{\sqrt{\beta_{x,m}\beta_{y,m}}}{4} \,
+    \frac{\exp{(i(\Delta\Phi_{x,mj} \mp \Delta\Phi_{y,mj}))}}{1-\exp({2\pi i (Q_x \mp Q_y}))}
+
+For people reading the code, the response matrices are first calculated like:
+
+::
+
+    |  Elements of interest (j) --> ... |
+    |Magnets (m)                        |
+    |  |                                |
+    |  v                                |
+    |  .                                |
+    |  .                                |
+    |  .                                |
+    |                                   |
+
+This avoids transposing all vectors individually in the beginning.
+At the end (of the calculation) the matrix is then transposed
+to fit the :math:`M \cdot \delta K` orientation.
+
+Also :math:`\Delta \Phi_{z,wj}` needs to be multiplied by :math:`2\pi` to be consistent.
+
+
+.. rubric:: References
+
+..  [#FranchiAnalyticformulasrapid2017]
+    A. Franchi et al.,
+    Analytic formulas for the rapid evaluation of the orbit response matrix
+    and chromatic functions from lattice parameters in circular accelerators
+    https://arxiv.org/abs/1711.06589
+
+.. [#TomasReviewlinearoptics2017]
+    R. Tomas, et al.,
+    'Review of linear optics measurement and correction for charged particle
+    accelerators.'
+    Physical Review Accelerators and Beams, 20(5), 54801. (2017)
+    https://doi.org/10.1103/PhysRevAccelBeams.20.054801
+
+"""
+import pickle
```

Comment:
any updates on `feather` or `parquette`? pickle is problematic because it often requires the same packes to be installed (with compatible versions) as when the object was pickled.